src/modules/auth/auth.service.test.ts

```diff
@@ -0,0 +1,45 @@
+import { ConfigService } from '@nestjs/config';
+import { PassportModule } from '@nestjs/passport';
+import { Test, TestingModule } from '@nestjs/testing';
+import Chance from 'chance';
+
+import { AuthService } from './auth.service';
+import { ApiKeyStrategy } from './strategy/api-key.strategy';
+
+describe('AuthService', () => {
+  let authService: AuthService;
+  const chance = new Chance();
+
+  beforeAll(async () => {
+    const app: TestingModule = await Test.createTestingModule({
+      imports: [PassportModule],
+      providers: [AuthService, ApiKeyStrategy, ConfigService],
+    }).compile();
+
+    authService = app.get<AuthService>(AuthService);
+  });
+
+  it('should be defined', () => {
+    expect(authService).toBeDefined();
+  });
+
+  it('should return `false` when API Key is invalid', () => {
+    const key = chance.word();
+    const result = authService.validateApiKey(key);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return `false` when API Key is not provided', () => {
+    const result = authService.validateApiKey('');
+
+    expect(result).toBe(false);
+  });
+
+  it('should return `true` when API Key is valid', () => {
+    const { API_KEY } = process.env;
```

Comment:
I didn't notice this in the MDM review, but actually `API_KEY` is going to be `undefined` here when we run `npm run unit-test` because we never load `API_KEY` into the environment variables.

The test is only passing because `ConfigService` and `AuthService`  also think the api key is `undefined` for the same
reason

I think we should do one of:
1. Change `AuthService` to have the config injected directly, instead of injecting `ConfigService`, like the other services in TFS do so that the test can supply the `API_KEY` in the `AuthService` constructor.
2. Mock the `ConfigService` in this test so that we can control the value that it returns for `get('app.apiKey')`. I think we should do this mocking by providing the `ConfigService` in the constructor directly, rather than trying to mock with `Test.createTestingModule`.

Personally I prefer 1

Reply:
this is updated now